### ISSUE TITLE
System.Numerics.IModulusOperators inconsistencies with naming and documentation

### ISSUE THREAD

**Issue by aloraman** [dotnet/runtime#75426]
#### TL;DR: 
The interface `System.Numerics.IModulusOperators` uses term `modulus` but means `modulo` (probably should use only `remainder`)

#### Long preamble:
Basically, this interface defines a constraint on implementing types to have `%` operator.
This operator is generally used to compute a _remainder_ of two values. E.g., `22 % 5` produces `2`, where `22` is _dividend_, `5` is _divisor_ and `2` is _remainder_. The related operation is _integer division_: `22 / 5` produces the _quotient_ `4`, so `4 * 5 + 2` produces `22` back. This pair of operations forms so called _Euclidian division_ in arithmetic. Note that terms **modulus** and **modulo** aren't used here. 

These terms come from _modular arithmetic_, where integers can be separated into certain classes (named _residues_) based on specific integer N, called **modulus**, so every two integers from the same class have N as a divisor of their difference, thus called _congruent_ **modulo** N. E.g., `22 ≡ 2 (mod 5)` - 22 is congruent 2 modulo 5. Obviously, these residue classes are tightly related to Euclidian division, all numbers in residue class "congruent 2 modulo 5" - {2,7,12,17,22,27...} - have the same remainder 2 when divided by 5. Since every residue class have infinite members, there are multiple ways to select a representative for each class. Generally, the set {0, 1, 2, ..., N − 1} is chosen, called the _least residue system modulo N_.

There are multiple ways to define result for integer division and remainder for negative numbers (based on actual Euclidian division or truncated/rounded/floored/ceiled division). Some are optimal for mathematical notation, others - for hardware implementation. It's no surprise there are a lot of differences with remainder computation between different programming languages, e.g., in C# `-22 % 5` produces `-2`, and in some other language `-22 mod 5` produces `3` (Some languages even have multiple options, e.g., Haskell with `div`/`quot` and `rem`/`mod`). Broadly speaking, if remainder operation produces result from the least residue system modulo divisor, it is called _modulo operation_ (`x mod y`) and the produced result is also called _modulo_. Note that it is **modulo**, not **modulus** - the divisor is modulus, not the operation and not the remainder. C# built-in remainder computation for integers `x % y` produces values from `[-abs(y)+1; abs(y)-1]` - not the least residue system, so it is not fair to call it modulo. 

#### Issue and questions:
At the moment, the documentation for `System.Numerics.IModulusOperators` states that it "Defines a mechanism for computing the **modulus** or remainder of two values", and the operator itself states that it "Divides two values together to compute their **modulus** or remainder". The actual **modulus** is `TOther right`, the produced thing is **modulo**. Therefore, I have these questions:

1. Is it possible to rename `IModulusOperators` to `IModuloOperators` or `IRemainderOperators`, or the ship has sailed already?
2. Should documentation be fixed (_modulus_ -> _modulo_), or am I wrong somewhere?
3. Should the term _modulus_/_modulo_ be dropped altogether, and only _remainder_ remain? The BCL implementations produce remainders, and the term _remainder_ for single operator covers _modulo_ anyway (I doubt there will be new `mod` operator in C#)

**Comment by tannergooding** [comment]
For historical reasons, the operator name for the C# `%` operator is `op_Modulus` in IL. This cannot change at this point and will always be this way.

API review reviewed possible alternative names and landed on matching the underlying operator name where possible. So `I*Operators` matching `op_*`. The one exception is `IBitwiseOperators` which covers `BitwiseAnd`, `BitwiseOr`, `ExclusiveOr`, and `OnesComplement` all together.

There are types throughout the ecosystem that return the "remainder" (such as most of the primitives/types built into the BCL) and types that do "modular arithmetic" and therefore wrap the value when reaching the `modulus`.

The name isn't ideal, but it is a historical inconvenience and one that we will never be able to fix. The documentation calls out `remainder` or `modulus` specifically so people are away that this may be the "standard remainder" or may instead be some other "modulus" related arithmetic (most typically that would be the `modulo` as the alternative). But there is no strict guarantee due to the wide variety of behaviors that different types may provide.

Other interfaces, such as `IBinaryInteger` or `IFloatingPointIeee754` require that `%` specifically be the "remainder".

**Comment by aloraman** [comment]
1. I see, it comes from CLI names for operator methods, ECMA-335 strikes again. It also explains the inconsistent naming between `IAdditionOperators` vs `IMultiplyOperators`.
2. I still don't like "return modulus" because modulus is an argument, not a result. Then again, I see that it follows from the same original decision to name the aforementioned method `op_Modulus`. Also, it seems modulo/modulus mix-up is a common thing (well, it's latin, not english) - so ordinary user would understand what this description actually conveys. 

Ultimately, I'm fine with it, I guess.
